phy/cluster/manual/session.py

```diff
@@ -440,12 +499,69 @@ def on_draw(event):
 
         return view
 
+    def _create_view_model(self, name, **kwargs):
+        vm_class = _VIEW_MODELS[name]
+        return vm_class(self.model, store=self.cluster_store, **kwargs)
+
+    # def _view_settings_name(self, view_model, name):
+    #     if isinstance(view_model, BaseViewModel):
+    #         view_model = view_model.view_name
+    #     return 'manual_clustering.' + view_model + '_' + name
+
+    # def _save_scale_factor(self, view_model):
+    #     name = self._view_settings_name(view_model, 'scale_factor')
+    #     if not name or not hasattr(view_model, 'scale_factor'):
+    #         return
+    #     sf = view_model.view.zoom * view_model.scale_factor
+    #     self.set_internal_settings(name, sf)
+
+    # def _load_scale_factor(self, view_name):
+    #     name = self._view_settings_name(view_name, 'scale_factor')
+    #     if not name:
+    #         return 1.
+    #     return self.get_internal_settings(name) or .01
+
     def show_waveforms(self):
-        return self._show_view(WaveformViewModel)
+        """Show a WaveformView and return a ViewModel instance."""
+
+        # Persist scale factor.
+        sf_name = 'manual_clustering.waveforms_scale_factor'
+        sf = self.get_internal_settings(sf_name) or .01
```

Comment:
Yep. The point is that the waveform normalization could be vastly different from a dataset to the next. So I just save whatever vertical channel height scaling factor the user is happy with, and I use that as a scaling factor for the waveforms. 

This will need to be refactored a bit when I implement saving of channel positions/scaling on the probe